test/badge.js

```diff
@@ -245,18 +245,23 @@ export async function downloadPdfFromBackpack(driver) {
 
     const dropdownButtons = await driver.findElements(By.css(
         'button[role="menuitem"]'));
-    const pdfExportButton = dropdownButtons[1];
+    const pdfExportButton = dropdownButtons[2];
     await pdfExportButton.click();
 
     const htmlEmbed = await driver.wait(until.elementLocated(
         By.css('embed[src^="blob:http"]')), defaultWait);
-    await driver.switchTo().frame(htmlEmbed);
+    // To avoid `No Such Frame` error
+    await driver.wait(until.ableToSwitchToFrame(htmlEmbed), defaultWait);
 
     await driver.wait(until.elementLocated(By.css('embed[src="about:blank"]')), defaultWait);
 
     await driver.switchTo().defaultContent();
-    const downloadButton = await driver.findElement(By.id(
-        'download-pdf-backpack'));
+    const downloadButton = await driver.findElement(By.id('download-pdf-backpack'));
+    // Wait for download button to be enabled
+    await driver.wait(async () => {
```

Comment:
It should be possible to do this wait more easily with [this](https://www.selenium.dev/selenium/docs/api/javascript/global.html#elementIsEnabled) wait

Reply:
Updated.